src/mintpy/save_hdfeos5.py

```diff
@@ -234,10 +243,19 @@ def get_output_filename(metadata, suffix=None, update_mode=False, subset_mode=Fa
 
     if subset_mode:
         print('Subset mode is enabled, put subset range info in output filename.')
-        lat1 = float(metadata['Y_FIRST'])
-        lon0 = float(metadata['X_FIRST'])
-        lat0 = lat1 + float(metadata['Y_STEP']) * int(metadata['LENGTH'])
-        lon1 = lon0 + float(metadata['X_STEP']) * int(metadata['WIDTH'])
+        if 'Y_FIRST' in metadata.keys():
+            lat1 = float(metadata['Y_FIRST'])
+            lon0 = float(metadata['X_FIRST'])
+            lat0 = lat1 + float(metadata['Y_STEP']) * int(metadata['LENGTH'])
+            lon1 = lon0 + float(metadata['X_STEP']) * int(metadata['WIDTH'])
+        elif 'mintpy.subset.lalo' in template.keys():
+            # for MiaplPy it would be preferrd to use miaplpy.subset.lalo but that is not available
```

Comment:
I agree. Then let's modify the `save_hdfeos5.py -t` option to 1) be able to take multiple template files, 2) read the miaplpy template file and pass it here?

Reply:
You  mean adding  `-t` for use outside `smallbaselineApp` with MiaplPy?  That would make sense.  But I think this is not only a MiaplPy issue. I believe I saw it occasionally for MintPy as well.

My preference would be to accept as is but make a note in the code that I will implement the -t option when I get a chance. After that we can switch back and we will see whether it really works.

The big question is what to do with Sara's geolocation correction and how to save the corrected coordinates in the S1 file. It would be good to implement this into MintPy but I don't see that Sara will have time for that any time soon. Unless you want to do it I probably will write a minsar script that would use  `save_hdfeos5.py -t`.

Reply:
I meant to modify `save_hdfeos5.py` to read both template files from mintpy and miaplpy, so that we could use `miaplpy.subset.lalo` as you wrote in the code comment.

> My preference would be to accept as is but make a note in the code that I will implement the -t option when I get a chance. After that we can switch back and we will see whether it really works.

Sounds fine to me.